user-service/src/main/java/com/rzheng/userservice/service/UserService.java

```diff
@@ -21,4 +21,8 @@ public List<User> getAllUsers() {
     public User addUser(User user) {
         return userRepository.save(user);
     }
+
+    public boolean doesEmailExist (String email) {
```

Comment:
You should have a base case checking if the email parameter is null or empty. In that case, just return false and don't need to waste system resource calling the repository layer and database.